homeassistant/util/location.py

```diff
@@ -49,14 +48,10 @@ async def async_detect_location_info(
     session: aiohttp.ClientSession,
 ) -> Optional[LocationInfo]:
     """Detect location information."""
-    data = await _get_ipapi(session)
-
-    if data is None:
-        data = await _get_ip_api(session)
+    data = await _get_ip_api(session)
```

Comment:
why not swap them around? that way we have still a fallback.

Also didn't our validation of the response work that it broke for you?